### DISCUSSION TITLE
Questions about the data reading flow in the Milvus blog

### DISCUSSION THREAD

**Discussion by caesarjuly** [milvus-io/milvus#24027] [Q&A and General discussion] [answered]
I'm reading the deep dive series of [blogs](https://milvus.io/blog/deep-dive-3-data-processing.md/#Read-data) on Milvus. I have a few questions about the reading process.
As we can see in the picture
1. The blog says, `Query requests are broadcast via DqRequestChannel to query nodes`. So I guess `DqRequestChannel` is a message channel or topic? This sounds like an async request
2. How do the DmChannels work in the data reading flow? There is no explanation in the read-data part; they are only mentioned in the write-data part.
3. About the gRPC result passing, is this a sync request/response?  The blog says, `Afterwards, query nodes need to aggregate the query results in both [sealed and growing segments](https://milvus.io/docs/v2.0.x/glossary.md#Segment). These aggregated results are passed on to proxy via gRPC.`
4. Is this an async DqRequestChannel + sync gRPC combination? It's the first time I see this kind of design. Please explain a little bit more about why.
Thank you 🙏

![image](https://github.com/milvus-io/milvus/assets/927521/dc61eebc-80cf-4534-82be-6e810ab59f80)

**Comment by xiaofan-luan** [comment, marked as answer]
Hi fan, Looks like you already know a. lot about Milvus.
1. Query used to send to DQL channel, but we've already remove that part to improve search latency (because MQ is usually too slow for sub 10 ms search)
2. New Search path: proxy -> shard leader -> other querynodes, through gprc.
3. proxy is responsible for checking search parameters, send request to all shard leaders  and reduce for the final result from all shards.
4. shard leader is responsible for maintain growing/sealed segment list, serve growing search, and route request to all querynodes holds historical segment, and reduce the result from all segments.
5. querynodes responsible for search historical segments, and reduce machine level results.

Does that make more sense to you?

**Reply by caesarjuly** [reply]
Thanks for the clear answer. 
Regarding `shard leaders`, it's the first time I meet this phrase, I find one doc [here](https://milvus.io/docs/replica.md#Shard-leader).
As it says, `A shard leader is the query node serving the streaming replica in a shard replica.`, I'm confused about the relationship between a shard and a query node. 
Is this correct?
1. One shard <-> one query node
2. One shard leader <-> one query node
3. One shard leader controls multiple shards <-> multiple query nodes

BTW, do we have new documents about the latest architecture? It seems like the blog is a bit old.
